File: 2. Bathroom Security/1.Solution.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    string moves;
    int arr[3][3] = {{1,2,3}, {4,5,6}, {7,8,9}};
    int curx = 1, cury = 1;
    int code = 0;

    for (int i = 0; i < 5; i++) { // no of lines of input
        getline(cin, moves);
        for (auto j : moves) {
            if (j == 'U' and cury > 0) cury--;
            else if (j == 'D' and cury < 2) cury++;
            else if (j == 'R' and curx < 2) curx++;
            else if (j == 'L' and curx > 0) curx--; 
        }
        code = code * 10 + arr[cury][curx];
    }

    cout << code;
}

File: 2016/1. No Time for a Taxicab/1.Solution.py
moves = input().split(', ')

x = y = 0
mx, my = 1, 0

for i in moves:
    j = int(i[1:])
    i = i[0]
    if i == 'R':
        mx, my = my, -mx
    else:
        mx, my = -my, mx
    x += mx * j
    y += my * j

print(abs(x) + abs(y))
